docs/prow/templates.md

```diff
@@ -0,0 +1,74 @@
+# Templates
+
+Jobs and Prow configuration are generated from templates that are in `templates` directory. There are following items there:
+
+- `config.yaml` - config file that describes files to generate
+- `templates` - directory to gather all template files
+
+### Config structure
+
+Config file is expected to have two keys:
+
+- `global` - a map of values available for all templates
+- `templates` - a list of files to generate
+
+Complete structure can be found [here](../../development/tools/cmd/rendertemplates/main.go).
+
+Other then that `config.yaml` uses YAML techniques for better organization, so normal YAML rules apply.
+
+### Template development
+
+A template will receive two objects as inputs:
+- `Values` which contains all `values` specified in its configuration in config file
+- `Global` which contains values specified under `global` in config file
+
+All functions from [`sprig`](https://github.com/Masterminds/sprig) library are available in templates. It is the same library that is used by helm, so if you know helm there's nothing new. Also some additional functions are available:
+- `releaseMatches <release> <since> <until>` - returns a bool value indicating if `release` fits in the range. You can use `nil` to remove one of the bounds. Example: `releaseMatches {{ $rel }} '1.2' '1.5'` will check if release `$rel` is younger then `1.2` and older then `1.5`.
```

Comment:
```suggestion
- `releaseMatches <release> <since> <until>` returns a boolean value indicating whether `release` fits in the range. Use `nil` to remove one of the bounds. For example: `releaseMatches {{ $rel }} '1.2' '1.5'` checks if the release `$rel` is younger than `1.2` and older than `1.5`.
```